2023-09-07: Handbook: Product dept: Wireframing  (FUTURE: Move to "Product groups")

noahtalerman opened the pull request:
- Add bullet about style improvement intake.

zhumo commented on handbook/product/README.md:
```diff
@@ -16,13 +16,15 @@ The product team is responsible for product design tasks like drafting [changes
-- Draft changes to the Fleet product that solve the problem specified in the issue. Constantly place yourself in the shoes of a user while drafting changes. Place these drafts in the appropriate Figma file in Fleet product project.
+- Draft changes to the Fleet product that solve the problem specified in the story. Constantly place yourself in the shoes of a user while drafting changes. Place these drafts in the appropriate Figma file in Fleet product project.
+
+- Be intentional about style improvements. These are made opportunistically as part of stories. Currently, Fleet doesn't have to capacity to take on style-only stories.
```
what is a "style improvement"?

zhumo replied:
we should not talk about "currently... capacity." Instead, it would be better if it were a clear policy. we don't do style only stories.

noahtalerman replied:
Thanks for the feedback!

Changed "style improvement" to "style changes" and gave some examples.

Updated language to make this a policy.

mikermcneil commented on handbook/product/README.md:
```diff
@@ -16,13 +16,15 @@ The product team is responsible for product design tasks like drafting [changes
-- Draft changes to the Fleet product that solve the problem specified in the issue. Constantly place yourself in the shoes of a user while drafting changes. Place these drafts in the appropriate Figma file in Fleet product project.
+- Draft changes to the Fleet product that solve the problem specified in the story. Constantly place yourself in the shoes of a user while drafting changes. Place these drafts in the appropriate Figma file in Fleet product project.
+
+- Be intentional about style changes that affect different parts of the product (e.g. button border-radius or modal width). These are made opportunistically as part of user stories. At Fleet we don't prioritize style-only user stories.
```
```suggestion
- Implemented behavior that doesn't match the style guide or wireframes, or looks odd at certain viewport widths, is a bug, and should be filed ASAP, as precisely as possible (with limited scope) so the root cause can be addressed quickly, usually in the current sprint before there is time for further expensive bugs or misunderstandings to accumulate.
- UI changes are not bugs if the current behavior is what was wireframed and documented.
- Be intentional about style guide changes that affect more than one area of the product (e.g. button border-radius or modal width). These changes are important and valued, and should be written as user stories like any other.  But they are also expensive.  They require extra drafting to carefully identify every place which needs to be changed ahead of time, and then extra quality assurance to check that all those places were correctly updated.
```

noahtalerman replied:
> These changes are important and valued, and should be written as user stories like any other.

Hey @mikermcneil this is not what @marko-lisica, @rachaelshaw, @mike-j-thomas, and I decided.

We decided to include style guide changes as part of stories. They don't get their own stories because these won't get prioritized in FF.

Let's discuss during the next design review. I'm closing this PR for now.

mikermcneil replied:
>  because these won't get prioritized in FF.

This is what needs to change.

mikermcneil replied:
>  because these won't get prioritized in FF.

This is what needs to change.  Like technical debt, UX debt will get prioritized if it drives value. 

An example of this is the recent work on "Queries", or the work we did at the top of 2023 to make the UI theme match the brand identity.

> Let's discuss during the next design review.
We can discuss at CEO office hours this evening, or sooner, making the decision and handbooking it ASAP.  This doesn't need to wait.  @zhumo @noahtalerman please join.

noahtalerman replied:
Heads up, I won't be able to join CEO office hours this evening. Mo is planning on joining.

mikermcneil replied:
Mo helped me see how I was somehow confused here.  Key part of my confusion was this sentence:
> At Fleet we don't prioritize style-only user stories.

Sorry for the confusion guys.

noahtalerman replied:
All good! Thanks for getting us on a call to clear that up.

noahtalerman commented:
> - Be intentional about changes to design components (e.g. button border-radius or modal width) because these are expensive. They'll require code changes and QA in multiple parts of the product. Propose changes to a design component as part of an already-prioritized user story instead of [making a new request](#making-a-request) in 🎁🗣 Feature Fest.

@mikermcneil @zhumo here's^ the updated language based on the feedback/discussion y'all had during CEO office hours last night. 

Here was the feedback:

> “At Fleet we don't prioritize style-only user stories” is the confusing part

I removed this language entirely. We want to make it clear that component updates are proposed, during design review, as part of already prioritized user stories. We don't bring component updates to feature fest.

> Make clear whether this is instruction for designer or engineer

Instructions are in the "Wireframing" section of the handbook. I think this clarifies that they're for the designer.

> Let’s write it in a way that directly addresses components.

Updated language to use "changes to design components" instead of "style changes."

Thanks for getting together last night.
